js/data/bucket/circle_bucket.js

```diff
@@ -40,19 +40,9 @@ CircleBucket.prototype.programInterfaces = {
             components: 4,
             type: 'Uint8',
             isLayerConstant: false,
-            value: (
-                'this._premultiplyColor(' +
-                    'layer.getPaintValue("circle-color", globalProperties, featureProperties),' +
-                    'layer.getPaintValue("circle-opacity", globalProperties, featureProperties)' +
-                ')'
-            ),
+            value: 'this._premultiplyColor(layer.getPaintValue("circle-color", globalProperties, featureProperties), 1)',
```

Comment:
I left the premultiplication outside the shader but I switch all shaders to separate color and opacity inputs in adb11ed and then removed the `additionalOpacity` parameter from `premultiply`.